synapse/handlers/federation.py

```diff
@@ -1922,15 +1922,15 @@ def add_display_name_to_third_party_invite(self, event_dict, event, context):
             original_invite = yield self.store.get_event(
                 original_invite_id, allow_none=True
             )
-        if not original_invite:
+        if original_invite:
+            display_name = original_invite.content["display_name"]
+            event_dict["content"]["third_party_invite"]["display_name"] = display_name
+        else:
             logger.info(
-                "Could not find invite event for third_party_invite - "
-                "discarding: %s" % (event_dict,)
+                "Could not find invite event for third_party_invite: %r",
+                event_dict
```

Comment:
Ok so we no longer discard a third_party_invite where we couldn't find the original invite event? Convince me that this is a sane thing to do (or better yet, write a comment to convince a future reader that this is a sane thing to do)